skipchain/struct.go

```diff
@@ -645,6 +645,45 @@ func (sbm *SkipBlockMap) GetFuzzy(id string) *SkipBlock {
 	return nil
 }
 
+// GetFuzzy searches for a block that resembles the given ID.
+// If there are multiple matching skipblocks, the first one is chosen. If none
+// match, nil will be returned.
+//
+// For every entry, the search is done in the following order:
+//  1. as prefix - if none is found
+//  2. as suffix - if none is found
+//  3. anywhere
+func (db *SkipBlockDB) GetFuzzy(id string) *SkipBlock {
+	match, err := hex.DecodeString(id)
+	if err != nil {
+		log.Error("Failed to decode " + id)
+		return nil
+	}
+	if len(match) == 0 {
+		log.Error("id is empty")
+		return nil
+	}
+
+	var sb *SkipBlock
+	db.View(func(tx *bolt.Tx) error {
```

Comment:
You have to make three passes through the keys... The idea is that a user will most probably use the prefix, so you should check for _all_ prefixes first. The next probably use-case is that the user gives the suffix, so you have to check for _all_ suffixes second. The most unprobably use-case is that the user takes a string in the middle, so you have to check for _all_ places in the end.

Come to think of it, for the `anywhere` case, the user might catch the hex-number in a nibble, so this will not be caught here. That's a good comment to make, that the `anywhere` check only checks for whole bytes, not for nibbles.

Reply:
Regarding the anywhere case, I'm wondering if that's really needed. It's difficult for me to imagine a case where the user would use that feature. Further, if the user makes a typo and an unintended block is matched, it may cause confusion further down the road. Should we only match the prefix and then the suffix?

Reply:
Why do we support anything other than prefix match? If this is just about usability, then just prefix match is enough to make it easy to type sc ids by hand.

Reply:
sometimes I like matching the end of a string - as it's not complicated and already in the code, I think it's a nice feature.